packages/shared/src/components/filters/FeedFilters.tsx

```diff
@@ -20,60 +26,78 @@ interface FeedFiltersProps {
+  action?: () => unknown;
+}
+
 export default function FeedFilters({
   isOpen,
   onBack,
 }: FeedFiltersProps): ReactElement {
+  const client = useQueryClient();
+  const [unblockItem, setUnblockItem] = useState<UnblockItem>();
+  const { user } = useContext(AuthContext);
   const { alerts, updateAlerts } = useContext(AlertContext);
   const { hasAnyFilter } = useFeedSettings();
-  const [hidden, setHidden] = useState(true);
-  const timeoutRef = useRef<number>();
+  const { flags } = useContext(FeaturesContext);
+  const shouldShowMyFeed = isFeaturedEnabled(Features.MyFeedOn, flags);
 
   useEffect(() => {
     if (isOpen) {
-      if (alerts?.filter && hasAnyFilter) {
+      if (alerts?.filter && hasAnyFilter && user) {
         updateAlerts({ filter: false });
       }
-
-      if (!hidden) {
-        return;
-      }
-
-      if (timeoutRef.current) clearTimeout(timeoutRef.current);
-      setHidden(false);
-      return;
     }
+  }, [isOpen, alerts, user, hasAnyFilter]);
 
-    if (hidden) {
-      return;
-    }
-
-    timeoutRef.current = window.setTimeout(() => setHidden(true), 300);
-    // eslint-disable-next-line consistent-return
-    return () => timeoutRef.current && clearTimeout(timeoutRef.current);
-  }, [isOpen, alerts, hasAnyFilter]);
+  const onCreate = () => {
+    // apply analytics
+    const key = getFeedSettingsQueryKey(user);
+    const { feedSettings } = client.getQueryData(key) as AllTagCategoriesData;
+    updateLocalFeedSettings(feedSettings);
+    window.location.replace(`${process.env.NEXT_PUBLIC_WEBAPP_URL}register`);
+  };
 
   return (
     <aside
       className={classNames(
         'fixed top-14 left-0 z-3 bottom-0 self-stretch bg-theme-bg-primary rounded-r-2xl border-t border-r border-theme-divider-primary overflow-y-auto',
         'transition-transform duration-200 ease-linear delay-100',
         isOpen ? 'translate-x-0' : '-translate-x-96 pointer-events-none',
-        hidden && 'invisible',
       )}
       style={{ width: asideWidth }}
     >
       <div
         className={classNames(
           menuItemClassNames,
-          'border-b border-theme-divider-tertiary',
+          'border-b border-theme-divider-tertiary flex-row justify-between',
         )}
       >
         <button onClick={onBack} type="button">
           <XIcon className="text-2xl -rotate-90 text-theme-label-tertiary" />
         </button>
+        {shouldShowMyFeed && (
+          <Button
+            className="btn-primary-avocado"
+            buttonSize="small"
+            icon={<PlusIcon className="mr-1" />}
+            onClick={onCreate}
+          >
+            Create
+          </Button>
+        )}
       </div>
-      <FilterMenu />
+      <FilterMenu onUnblockItem={setUnblockItem} />
+      {unblockItem && (
+        <UnblockModal
+          item={unblockItem}
+          isOpen={!!unblockItem}
+          onConfirm={unblockItem.action}
+          onRequestClose={() => setUnblockItem(null)}
+        />
+      )}
```

Comment:
Why do you introduce this change?

Reply:
Oh, I might have missed this on my initial plan on how to manage the data from the feed filters.